spring-kafka/src/main/java/org/springframework/kafka/config/AbstractKafkaListenerContainerFactory.java

```diff
@@ -379,13 +380,27 @@ else if (this.autoStartup != null) {
 						instance.getContainerProperties()::setConsumerProperties);
 	}
 
+	/**
+	 * {@inheritDoc}
+	 * @deprecated in favor of {@link #createContainer(TopicPartitionOffset[])}
+	 */
+	@Deprecated
+	@SuppressWarnings("deprecation")
+	@Override
+	public C createContainer(final Collection<org.springframework.kafka.support.TopicPartitionInitialOffset> topicPartitions) {
+		return createContainer(topicPartitions.stream()
+				.map(TopicPartitionOffset::fromTPIO)
```

Comment:
Maybe such a mapper should be in an opposite place - `TopicPartitionInitialOffset.toTopicPartitionOffset()`?

Even if this is going to cause some class tangle, that mapper is going to disappear together with deleted deprecated class.